aten/src/ATen/native/cpu/Vec256.h

```diff
@@ -3,6 +3,9 @@
 #if defined(_MSC_VER)
 /* Microsoft C/C++-compatible compiler */
 #include <intrin.h>
+#if _MSC_VER <= 1900
+#define _mm256_extract_epi64(X, Y) (((uint64_t*)&X)[Y])
```

Comment:
Whaaat? Are you sure this is ok? Casting __mm256 to a pointer to uint sounds like a really bad idea. It’s a compiler specific type, how is it implemented in MSC?

Reply:
It is defined as the following.
```
typedef union  __declspec(intrin_type) __declspec(align(32)) __m256i {
    __int8              m256i_i8[32];
    __int16             m256i_i16[16];
    __int32             m256i_i32[8];
    __int64             m256i_i64[4];
    unsigned __int8     m256i_u8[32];
    unsigned __int16    m256i_u16[16];
    unsigned __int32    m256i_u32[8];
    unsigned __int64    m256i_u64[4];
} __m256i;
```
You are right and I'll revert this.

Reply:
But actually it is a solution posted in this [post](https://github.com/WojciechMula/sse-popcount/issues/17).